cmd/root.go

```diff
@@ -57,18 +70,19 @@ var rootCmd = &cobra.Command{
 		go detectNewVersion()
 
 		return nil
-	},
+	}
 }
 
 // hold a pointer to the global lifecycle controller so that commands could output messages and exit properly
 var glcm = common.GetLifecycleMgr()
 
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
-func Execute(azsAppPathFolder, logPathFolder string) {
+func Execute(azsAppPathFolder, logPathFolder string, steStartup steStartupFunc) {
 	azcopyAppPathFolder = azsAppPathFolder
 	azcopyLogPathFolder = logPathFolder
 
+	rootCmd.PersistentPreRunE = createPersistentPreRunE(steStartup)
```

Comment:
So, my understanding of what's going on here is that we're feeding our steStartup function from main.go into the root cmd, baking it into a pre-run function, and then letting cobra ingest it here. Is it feasible to just cut this down to the last step? That might proactively clean up some of the wild-goose-chaseyness of our codebase.

Reply:
Yes, we could possibly move it all to there.  Interested in what @zezha-msft thinks....

Reply:
Agreed. Having the initializing code all in one place would help a lot with the readability.